cmd/profiling.go

```diff
@@ -0,0 +1,91 @@
+/*
+This file is part of REANA.
+Copyright (C) 2022 CERN.
+
+REANA is free software; you can redistribute it and/or modify it
+under the terms of the MIT License; see LICENSE file for more details.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"runtime/pprof"
+
+	"github.com/spf13/pflag"
+)
+
+var (
+	profileMode   string
+	profileOutput string
+)
+
+func addProfilerFlags(flags *pflag.FlagSet) {
+	flags.StringVar(
+		&profileMode,
+		"profile",
+		"none",
+		"Enable profiling. One of (none|cpu|heap)",
+	)
+	flags.StringVar(
+		&profileOutput,
+		"profile-output",
+		"profile.pprof",
+		"Name of the file to write the profile to",
+	)
+}
+
+func setupProfiler() error {
+	var (
+		f   *os.File
+		err error
+	)
+	switch profileMode {
+	case "none":
+		return nil
+	case "cpu":
+		f, err = os.Create(profileOutput)
+		if err != nil {
+			return err
+		}
+		err = pprof.StartCPUProfile(f)
```

Comment:
Works well :+1: Although the sample coverage does not look great, as we discussed IRL. E.g. for one of my tests:

```
Duration: 1.06s, Total samples = 30ms ( 2.83%)
Showing nodes accounting for 30ms, 100% of 30ms total
      flat  flat%   sum%        cum   cum%
      10ms 33.33% 33.33%       10ms 33.33%  encoding/json.(*decodeState).rescanLiteral
      10ms 33.33% 66.67%       10ms 33.33%  encoding/json.stateInString
      10ms 33.33%   100%       10ms 33.33%  runtime.(*spanSet).push
         0     0%   100%       20ms 66.67%  encoding/json.(*Decoder).Decode
         0     0%   100%       10ms 33.33%  encoding/json.(*Decoder).readValue
         0     0%   100%       10ms 33.33%  encoding/json.(*decodeState).array
...
```

If parametrising sample taking via a new CLI option wouldn't help much here, then we could merge "as is"